src/codewhisperer/util/codewhispererStates.ts

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export class CodeWhispererStates {
```

Comment:
should `vsCodeState` be merged into this class? https://github.com/aws/aws-toolkit-vscode/blob/a5031452fb1333429a6dcf26da19492e432c2612/src/codewhisperer/models/model.ts#L33

there is also `globalStateUtil.ts`

Reply:
This `CodeWhispererStates` is aiming for a CW session(per trigger) scope and this is only part of the progress to group some of the variables that have the same scope together. Later on we will call `CodeWhispererStates.reset()` for every CW session in which case I think it's more appropriate to separate the these states vs the `vsCodeState` because they are global scopes. I will change it class name to be `CodeWhispererSessionStates` to make it clear.